datasets/xtreme_s/xtreme_s.py

```diff
@@ -0,0 +1,780 @@
+                line = line.strip()
+                _id, transcript = line.split("\t")
+
+                if sub_folder != "" and _id not in all_ids:
+                    # filter-out audios not contained in the 9/10h version
+                    continue
+
+                audio_file = f"{_id}.flac"
+                speaker_id, chapter_id = [int(el) for el in _id.split("_")[:2]]
+
+                yield key, {
+                    "path": os.path.join(data_dir, "audio", str(speaker_id), str(chapter_id), audio_file),
+                    "audio": os.path.join(data_dir, "audio", str(speaker_id), str(chapter_id), audio_file),
+                    "transcription": transcript,
+                }
+                key += 1
+
+    # Voxpopuli
+    def _voxpopuli_split_generators(self, dl_manager):
+        # TODO(PVP) - need to add warning here that even for a single
+        # language ALL files need to be downloaded because all langs archive_path
+        # mixed in single audio files
+        if hasattr(dl_manager, "download_config"):
+            # last file is text file
+            dl_manager.download_config.num_proc = len(_VOXPOPULI_AUDIO_URLS)
+
+        extracted_audio_data_list = dl_manager.download_and_extract(self.config.data_urls[:-1])
+        extracted_audio_data_dict = {
+            str(i + 2009): extracted_audio_data_list[i] for i in range(len(extracted_audio_data_list))
+        }
+
+        extracted_text_data = dl_manager.download_and_extract(self.config.data_urls[-1].format(self.config.lang_name))
+
+        # TODO(PVP) - add print statement that this takes a while or into tqdm
+        # split audio data and save in cached extracted folders so that it
+        # only has to be done once
+        split_audio_data = self._voxpopuli_split_audio(extracted_audio_data_dict, extracted_text_data)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "audio_data": split_audio_data,
+                    "text_data": extracted_text_data,
+                    "split": "train",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={
+                    "audio_data": split_audio_data,
+                    "text_data": extracted_text_data,
+                    "split": "dev",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={
+                    "audio_data": split_audio_data,
+                    "text_data": extracted_text_data,
+                    "split": "test",
+                },
+            ),
+        ]
+
+    def _voxpopuli_split_audio(self, audio_data_dict, text_data, sub_folder="original"):
+        """This function is heavily inspired from https://github.com/facebookresearch/voxpopuli/blob/main/voxpopuli/get_asr_data.py"""
```

Comment:
Feel free to add comments to explain why you need this, and add some logging since it can take some time